### PR TITLE
Sign Up Flow UI

### PR DESCRIPTION
**Pull request by guytepper**

### CODE REVIEW

File: src/views/LeaderRequest.js [hunk 1]
```diff
@@ -1,56 +1,74 @@
...
 
+  useEffect(() => {
+    const { protest: protestId } = queryString.parse(window.location.search);
+
+    if (!protestId) {
+      console.error('Missing protest ID, cannot assign user to a protest');
+    }
+
+    getProtestById(protestId).then((protestData) => {
+      console.log(protestData);
```
Comment (ranyitz) [inline src/views/LeaderRequest.js]: extra console.log
```suggestion
```

File: src/views/SignUp.js [hunk 2]
```diff
@@ -30,39 +29,48 @@ export default function SignUp(props) {
...
-        if (returnUrl) {
-          history.push(returnUrl);
+      .then((result) => {
+        if (!result) {
+          setStage(stages.BEFORE_FACEBOOK_AUTH);
+          return;
         }
+
+        const userData = extractUserData(result);
+        console.log(result, userData);
```
Comment (ranyitz) [inline src/views/SignUp.js]: extra console.log
```suggestion
```

**Review by ranyitz — APPROVED** [review]
Looking good

File: src/api/index.js [hunk 3]
```diff
@@ -161,12 +166,12 @@ export async function getProtestsForLeader(uid) {
   var protestsRef = firestore.collection('protests');
   var query = protestsRef.where('roles.leader', 'array-contains', uid);
 
-  const querySnapshot = await query.get()
+  const querySnapshot = await query.get();
   const protests = [];
```
Comment (webernir) [inline src/api/index.js:170]:
```suggestion
return querySnapshot.map((doc) => ({ id: doc.id, ...doc.data() }));
```

**Pull request merged**